omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java

```diff
@@ -0,0 +1,148 @@
+        if (logSource == null && identifier == null && comment == null && dateFrom == null && dateTo == null
+                && user == null) {
+            return new EmptySearchResult();
+        }
+        logEntries = identifierSourceService.getLogEntries(logSource, dateFrom, dateTo, identifier, user, comment);
+
+        return new NeedsPaging<LogEntry>(logEntries, context);
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        DelegatingResourceDescription description = null;
+        if (rep instanceof RefRepresentation) {
+            description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("display");
+            description.addProperty("identifier");
+            description.addSelfLink();
+        } else if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+            description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("source");
+            description.addProperty("identifier");
+            description.addProperty("comment");
+            description.addProperty("generatedBy");
+            description.addProperty("dateGenerated");
+            description.addSelfLink();
+            if (rep instanceof DefaultRepresentation) {
+                description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            }
+        }
+        return description;
+    }
+
+    @PropertyGetter("uuid")
+    public String getUuid(LogEntry logentry) {
+        return logentry.getIdentifier();
+    }
+
+    @PropertyGetter("source")
+    public String getSource(LogEntry logentry) {
+        return logentry.getSource().getUuid();
+    }
+
+    @PropertyGetter("generatedBy")
+    public String getGeneratedBy(LogEntry logentry) {
+        return logentry.getGeneratedBy().getUuid();
+    }
+
+    @PropertyGetter("display")
+    public String getDisplay(LogEntry logentry) {
+        return logentry.getSource().toString();
+    }
+
+    @Override
+    public LogEntry newDelegate() throws ResourceDoesNotSupportOperationException {
+        throw new UnsupportedOperationException("method not supported");
+    }
+
+    @Override
+    public LogEntry save(LogEntry delegate) throws ResourceDoesNotSupportOperationException {
+        throw new UnsupportedOperationException("save not supported");
+    }
+
+    @Override
+    public LogEntry getByUniqueId(String uniqueId) {
+        return Context.getService(IdentifierSourceService.class).getLogEntries(null, null, null, uniqueId, null, null).get(0);
+    }
+
+    @Override
+    protected void delete(LogEntry delegate, String reason, RequestContext context)
+            throws ResourceDoesNotSupportOperationException {
+        throw new UnsupportedOperationException("delete not supported");
```

Comment:
Should be ResourceDoesNotSupportOperationException

Reply:
Fixed this too, thanks